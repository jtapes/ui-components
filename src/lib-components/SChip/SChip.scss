@import '../../theme/styles/variables';
@import '../../theme/styles/mixins';
@import '../../theme/styles/reset';
@import '../../theme/styles/fonts';
@import '../../theme/styles/global';

.s-chip {
  display: inline-flex;
  align-items: center;
  border: none;
  border-radius: $size--24;
  height: $size--48;
  color: $color--primary;
  background-color: $color--background;
  transition: $animate--time;
  cursor: pointer;

  &__content {
    display: flex;
    padding: 0 $size--20;
  }

  &__text {
    display: inline-block;
    overflow: hidden;
    max-width: 164px;
    white-space: nowrap;
    text-overflow: ellipsis;
  }

  &__count {
    margin-left: $size--12;
    color: $color--font-muted;
  }

  &__close {
    display: flex;
    align-items: center;
    justify-content: center;
    margin: $size--8;
    border-radius: 50%;
    width: $size--32;
    height: $size--32;
    color: $color--primary;
    background-color: $color--white;
  }

  &:not(&--disabled, &--closable):hover {
    color: $color--primary-accent;
    background-color: $color--complimentary-bg-light;
  }

  &--active {
    color: $color--white;
    background-color: $color--primary-dark;
  }

  &--disabled {
    color: $color--border;
    background-color: $color--background;
    pointer-events: none;
  }

  &--closable {
    .s-chip__content {
      padding-right: 0;
      padding-left: $size--20;
    }

    &:hover {
      box-shadow: 0 0 1px 1px $color--border;
      background-color: $color--white;

      .s-chip__close {
        color: $color--primary-accent;
        background-color: $color--complimentary-bg-light;
      }
    }
  }
}
