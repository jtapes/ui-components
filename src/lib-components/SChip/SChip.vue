<template>
  <div
    class="s-chip text-sm"
    :class="{
      's-chip--closable': closable,
      's-chip--active': active,
      's-chip--disabled': disabled,
    }"
    @click="handleChipClose"
  >
    <div class="s-chip__content">
      <div class="s-chip__text">
        <!-- @slot Слот для контента  -->
        <slot>
          {{ label }}
        </slot>
      </div>
      <span v-if="count" class="s-chip__count">{{ count }}</span>
    </div>
    <div v-if="closable" class="s-chip__close">
      <svg-icon name="times-24-32" class="icon-32" />
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'
import SvgIcon from '@/lib-components/SvgIcon/SvgIcon.vue'

/** Компонент используется для передачи небольших фрагментов информации */
@Component({
  components: {
    SvgIcon,
  }
})
export default class SChip extends Vue {
  /** Контент внутри chip */
  @Prop(String)
  label?: string

  /** Добавление кнопки "закрыть" */
  @Prop(Boolean)
  closable?: boolean

  /** Отвечает за активность */
  @Prop(Boolean)
  active?: boolean

  /** Число, которое отображается внутри Chip */
  @Prop(Number)
  count?: number

  /** Disabled свойство компонента */
  @Prop(Boolean)
  disabled?: boolean

  handleChipClose(): void {
    if (this.closable) {
      this.$emit('close')
    }
  }
}
</script>
