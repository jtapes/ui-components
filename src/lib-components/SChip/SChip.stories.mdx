import { action } from '@storybook/addon-actions'
import { config, withDesign } from 'storybook-addon-designs'
import { withTests } from '@storybook/addon-jest';
import {
  Meta,
  Story,
  ArgsTable,
  Description,
  Source,
  Props,
  Preview,
} from '@storybook/addon-docs'
import results from '/.jest-test-results.json';
import SChip from '@/lib-components/SChip/SChip.vue'

<Meta
  title="DESIGN SYSTEM/Atom/Chip"
  component={SChip}
  decorators={[withDesign, withTests({results})]}
  argTypes={{
    label: { control: 'text' },
    closable: { control: 'boolean' },
    active: { control: 'boolean' },
    disabled: { control: 'boolean' },
    count: { control: 'number' },
  }}
  parameters={{
    design: config({
      type: 'figma',
      url:
        'https://www.figma.com/file/IIeFkxNoHGcaqNHlVeNrhW/MVP-3-Market-UI-Kit?node-id=846%3A1195',
    }),
    docs: {
      source: {
        type: 'code',
      },
    },
  }}
/>

# Chip

<Description of={SChip} />

export const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { SChip },
  template:
    `<div style="display: inline-block">
      <s-chip @close="handleClick" :label="label" :closable="closable" :active="active" :disabled="disabled" :count="count" />
    </div>`,
  methods: {
    handleClick() {
      action('close')()
    }
  }
})

<Preview withToolbar withSource="none">
  <Story name="SChip" args={{ label: '18 мм', closable: false }}>
    {Template.bind({})}
  </Story>
</Preview>

## Пример

<Source
  language="html"
  code={`
    <s-chip closable>
      18 мм
    </s-chip>
  `}
/>

<ArgsTable of={SChip} />
