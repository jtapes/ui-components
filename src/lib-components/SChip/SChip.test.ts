import { mount } from '@vue/test-utils'
import SChip from '@/lib-components/SChip/SChip.vue'

describe('SChip', () => {
  test('Is Vue snapshot', () => {
    const wrapper = mount(SChip, {
      propsData: {
        closable: true,
      },
      stubs: ['svg-icon', 'nuxt-link'],
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  test('Is emitted by $emit()', async () => {
    const wrapper = mount(SChip, {
      propsData: {
        closable: true,
      },
      stubs: ['svg-icon', 'nuxt-link'],
    })

    const buttonClose = wrapper.find('.s-chip')
    await buttonClose.trigger('click')

    expect(wrapper.emitted('close')).toBeTruthy()
  })
  test('Count of chip exist count prop', () => {
    const count = 5
    const wrapper = mount(SChip, {
      propsData: {
        count,
      },
      stubs: ['svg-icon', 'nuxt-link'],
    })

    expect(wrapper.find('.s-chip__count').text()).toBe(String(count))
  })
})
