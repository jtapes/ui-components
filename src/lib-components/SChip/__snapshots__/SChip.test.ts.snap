// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SChip Is Vue snapshot 1`] = `
<div
  class="s-chip text-sm s-chip--closable"
>
  <div
    class="s-chip__content"
  >
    <div
      class="s-chip__text"
    >
      
        
      
    </div>
     
    <!---->
  </div>
   
  <div
    class="s-chip__close"
  >
    <svg-icon-stub
      class="icon-32"
      name="times-24-32"
    />
  </div>
</div>
`;
